tools/guid_generator/setup.py

```diff
@@ -24,5 +24,5 @@
     author='Travis Welch',
     packages=find_packages(),
     install_requires=['strictyaml==1.4.2'],
-    python_requires='>=3.9',
+    python_requires='>=3.11',
```

Comment:
those are ok to keep at >3.9
since the 3.9 is supported until 2025
Please update only the github actions to make the tests go faster, since 3.11 introduces performance improvements 
https://www.python.org/downloads/